parser/parser.go

```diff
@@ -7,5 +7,10 @@ import (
 
 // NewAST parses Flux query and produces an ast.Program
 func NewAST(flux string) (*ast.Program, error) {
-	return fastparser.NewAST([]byte(flux)), nil
+	n := fastparser.NewAST([]byte(flux))
+	if ast.Check(n) > 0 {
+		errs := ast.GetErrors(n)
+		return nil, errs[0]
```

Comment:
This should not return an error here. We want to be able to return invalid ASTs from the parser. As such I do not think it should run the Check step here either, rather if you need to Check an AST you do so explicitly. In other words assume an ast Node has not been Checked unless you did it explicitly.

Reply:
Done.